service/paypal.go

```diff
@@ -97,6 +103,14 @@ func (pp *PayPalService) CreatePaymentAndGenerateNextURL(req *http.Request, paym
 	return nextURL, Success, nil
 }
 
+func (pp *PayPalService) GetOrderDetails(id string) (*paypal.Order, error) {
```

Comment:
I would put this logic in the `GetPaymentDetails` function. Rather than returning a `*paypal.Order` type we can return the `*models.PaymentDetails` type and transform the data returned by PayPal into the data expected in that struct. The only thing we use from this response is the `status` anyway which is already contained there. 

This will mean we can remove the empty `GetOrderDetails` function from GovPay as well.

Reply:
The main difference with this function (and the reason we kept it separate) is that the Paypal SDK requires only the orderID to make the request, rather than *models.PaymentResourceRest which is used by the GovPay function and is used to make the GovPay call.

Reply:
GovPay also only requires an ID to make the request to check the payment status but it's still a good practice to pass in the entire model as we can pass it in as a pointer so it doesn't effect the performance. Also if we ever need to update the method later we have all the data we need already present.

That being said this will need to be updated later anyway as we're currently not saving the `ExternalPaymentStatusURI` in Mongo when we create the PayPal session, which we should be retrieving for this part of the callback journey and using for this check. Not saving this URI might cause some problems for support if they need to investigate any problems against PayPal with payments being taken. 

I'm happy to accept for the time being because it's not a massive issue, I'm just wary of the interface becoming bogged down with lots of external-payment-provider specific services which are doing very similar things